runtime/near-vm/compiler-singlepass/src/codegen_x64.rs

```diff
@@ -365,10 +377,12 @@ impl<'a> FuncGen<'a> {
         if cost_location == Location::Imm32(0) {
             return; // skip, which we must do because emit_add optimizes out the add 0 which leaves CF clobbered otherwise
         }
-        assert!(
-            matches!(cost_location, Location::Imm32(_) | Location::GPR(_)),
-            "emit_gas can take only an imm32 or a gpr argument"
-        );
+
+        match cost_location {
+            Location::Imm32(v) if v <= i32::MAX as u32 => {}
+            Location::GPR(_) => {}
+            _ => panic!("emit_gas can take only a imm32 < 0xFFF_FFFF or a gpr argument"),
+        }
```

Comment:
I guess this needs feature flagging to avoid the unfixed version from panicking.

(Unless you want o change the `emit_gas_cons` to reproduce the bug in other ways, like manually sign extending to a GPR or something along those lines... which sounds horrifying.)